lib/friends/introvert.rb

```diff
@@ -212,6 +212,27 @@ def add_nickname(name:, nickname:)
       @output << "Nickname added: \"#{friend}\""
     end
 
+    # Add an alias to an existing location.
+    # @param name [String] the name of the location
+    # @param nickname [String] the alias to add to the location
+    # @raise [FriendsError] if 0 or 2+ location match the given name
+    # @raise [FriendsError] if the alias is already taken
+    def add_location_alias(name:, nickname:)
+      raise FriendsError, "Expected \"[Location Name]\" \"[Alias]\"" if name.empty?
+      raise FriendsError, "Alias cannot be blank" if nickname.empty?
+
+      if @locations.any? { |location| location.nicknames.include?(nickname) }
```

Comment:
Thanks for adding this check! A few small comments:

1. Since we'll want to know which location matches, can we do the lookup once instead of do `any?` and then the `select`?
2. Can we also check that no location alias matches another location's _name_? (And similarly add that check in `add_location`.)
3. `find` is a more efficient alternative to `select { ... }.first`
4. Should we do this matching case-insensitively? (I realize that `add_location` and `add_friend` don't currently use case-insensitive matching but I should probably change that.) I'm happy to move that to a future PR if you'd prefer.

Concretely, I'm imagining something like:

```ruby
collision = @locations.find do |loc|
  loc.name.casecmp(nickname) == 0 || loc.nicknames.any? { |n| n.casecmp(nickname) == 0 }
end

if collision
  raise FriendsError, ...
end
```

And actually if we're doing this check in both `add_location` and `add_location_alias` we should probably extract it to a helper method.

Thoughts?

Reply:
1. Yes, of course. The double checkup is not needed. 
2. Right, this could also happen and is a good point. 
3. I see. Also it is more descriptive for that case. Somehow only `find_index` was listed in the documentation i had open. 
4. The methods that match a name in an activity use a case-insensitiv regex. Because of that i think it would only make sense to do the check match while adding a new one case-insensitiv , too. I think another PR would be better. 


Yes after using that in both methods it should be exist a helper method.

That look great also seems to work as expected.